app/controllers/questions_controller.rb

```diff
@@ -26,6 +26,16 @@ def destroy
     flash[:notice] = "Question successfully deleted" 
   end
 
+  def clear_votes
+    question = Question.find(params[:id])
+    if question.choices.map{ |choice| choice.votes.destroy_all }
+      redirect_to question_path(question)
+      flash[:notice] = "Votes successfully cleared"
+    else
+      flash[:alert] = "Error clearing votes"
```

Comment:
It's not absolutely perfect because, even though no-one can access the button in the view if there are no votes, in theory someone could run the method somehow and it would still say 'votes cleared' even when there weren't any votes, but all-in-all it's a moot point I think.